kinto_portier/views.py

```diff
@@ -0,0 +1,137 @@
+    nonce = persist_nonce(request)
+    form_url = ('{broker_uri}auth?{query_args}')
+    broker_uri = portier_conf(request, 'broker_uri')
+
+    query_args = urlencode({
+        'login_hint': request.validated['email'],
+        'scope': portier_conf(request, 'requested_scope'),
+        'nonce': nonce,
+        'response_type': 'id_token',
+        'response_mode': 'form_post',
+        'client_id': '{scheme}://{host}'.format(scheme=request.registry.settings['http_scheme'],
+                                                host=request.registry.settings['http_host']),
+        'redirect_uri': request.route_url(verify.name),
+    })
+
+    return httpexceptions.HTTPFound(location=form_url.format(broker_uri=broker_uri,
+                                                             query_args=query_args))
+
+
+class PortierVerifyQuerystring(colander.MappingSchema):
+    error = colander.SchemaNode(colander.String(), missing=colander.drop)
+    error_description = colander.SchemaNode(colander.String(), missing=colander.drop)
+
+
+class PortierVerifyPayload(colander.MappingSchema):
+    id_token = colander.SchemaNode(colander.String())
+
+
+class PortierVerifyRequest(colander.MappingSchema):
+    body = PortierVerifyPayload()
+    querystring = PortierVerifyQuerystring()
+
+    def deserialize(self, cstruct=colander.null):
+        appstruct = super(PortierVerifyRequest, self).deserialize(cstruct)
+        error = appstruct['querystring'].get('error')
+        if error:
+            msg = 'Broker error (%s)' % error
+            desc = appstruct['querystring'].get('error_description')
+            if desc:
+                msg += ': %s' % desc
+            self.raise_invalid(msg)
+        return appstruct
+
+
+@verify.post(schema=PortierVerifyRequest(), permission=NO_PERMISSION_REQUIRED,
+             validators=(colander_validator,))
+def portier_verify(request):
+    """Helper to redirect client towards Portier login form."""
+    broker_uri = portier_conf(request, 'broker_uri')
+    token = request.validated['body']['id_token']
+    audience = '{scheme}://{host}'.format(scheme=request.registry.settings['http_scheme'],
+                                          host=request.registry.settings['http_host']),
+
+    try:
+        email, stored_redirect = get_verified_email(
+            broker_url=broker_uri,
+            token=token,
+            audience=audience,
+            issuer=broker_uri,
+            cache=request.registry.cache)
+    except RuntimeError as exc:
+        error_details = {
+            'name': 'id_token',
+            'location': 'body',
+            'description': 'Portier token validation failed: %s' % exc
+        }
+        raise_invalid(request, **error_details)
+
+    hmac_secret = request.registry.settings['userid_hmac_secret']
+    user_token = utils.hmac_digest(hmac_secret, email)
+    request.registry.cache.set('portier:' + user_token, email,
+                               timedelta(days=1).seconds)
```

Comment:
we should probably have a setting for this as well instead of hardcoded timedelta